packages/discovery-provider/src/queries/get_authorization.py

```diff
@@ -0,0 +1,30 @@
+from src.models.grants.grant import Grant
+from src.utils import db_session
+from src.utils.auth_middleware import recover_authority_from_signature_headers
+
+
+def is_authorized_request(user_id: int):
```

Comment:
unique from `is_active_manager` in that it:
1. Doesn't have to be a user (can be dev app)
2. Doesn't introduce a circular dep (due to the "populate" helpers)
3. Recovers the user authority from the signature headers

Reply:
this guy seems pretty testable in isolation